File: app/src/main/java/com/oddlyspaced/calci/mathtype/enums/MTInlineOperatorType.java
package com.sparkappdesign.archimedes.mathtype.enums;

import com.sparkappdesign.archimedes.mathexpression.enums.METrigonometricFunctionType;
/* loaded from: classes.dex */
public enum MTInlineOperatorType {
    None,
    Plus,
    Minus,
    Dot,
    Division,
    Power,
    SquareRoot,
    NaturalLogarithm,
    Equals,
    EngineeringExponent,
    Percent,
    PerMil,
    Sine,
    Cosine,
    Tangent,
    ArcSine,
    ArcCosine,
    ArcTangent,
    Cosecant,
    Secant,
    Cotangent,
    ArcCosecant,
    ArcSecant,
    ArcCotangent,
    HyperbolicSine,
    HyperbolicCosine,
    HyperbolicTangent,
    ArcHyperbolicSine,
    ArcHyperbolicCosine,
    ArcHyperbolicTangent;

    public static boolean isTrigonometric(MTInlineOperatorType type) {
        return type.ordinal() >= Sine.ordinal() && type.ordinal() <= ArcHyperbolicTangent.ordinal();
    }

    public static MTInlineOperatorType fromMETrigonometricFunctionType(METrigonometricFunctionType type) {
        switch (type) {
            case Sine:
                return Sine;
            case Cosine:
                return Cosine;
            case Tangent:
                return Tangent;
            case ArcSine:
                return ArcSine;
            case ArcCosine:
                return ArcCosine;
            case ArcTangent:
                return ArcTangent;
            default:
                throw new IllegalArgumentException("Invalid enum argument: " + type);
        }
    }
}

File: app/src/main/java/com/oddlyspaced/calci/mathtype/enums/MTInlineOperatorType.kt
package com.oddlyspaced.calci.mathtype.enums

import com.oddlyspaced.calci.mathexpression.enums.METrigonometricFunctionType

enum class MTInlineOperatorType {
    None,
    Plus,
    Minus,
    Dot,
    Division,
    Power,
    SquareRoot,
    NaturalLogarithm,
    Equals,
    EngineeringExponent,
    Percent,
    PerMil,
    Sine,
    Cosine,
    Tangent,
    ArcSine,
    ArcCosine,
    ArcTangent,
    Cosecant,
    Secant,
    Cotangent,
    ArcCosecant,
    ArcSecant,
    ArcCotangent,
    HyperbolicSine,
    HyperbolicCosine,
    HyperbolicTangent,
    ArcHyperbolicSine,
    ArcHyperbolicCosine,
    ArcHyperbolicTangent;

    companion object {
        fun isTrigonometric(type: MTInlineOperatorType): Boolean {
            return type.ordinal >= Sine.ordinal && type.ordinal <= ArcHyperbolicTangent.ordinal
        }

        fun fromMETrigonometricFunctionType(type: METrigonometricFunctionType): MTInlineOperatorType {
            return when (type) {
                METrigonometricFunctionType.Sine -> Sine
                METrigonometricFunctionType.Cosine -> TODO()
                METrigonometricFunctionType.Tangent -> TODO()
                METrigonometricFunctionType.ArcSine -> TODO()
                METrigonometricFunctionType.ArcCosine -> TODO()
                METrigonometricFunctionType.ArcTangent -> TODO()
            }
        }
    }
}